src/index.ts

```diff
@@ -1,19 +1,34 @@
 import Config from './config';
-import { Client, Intents } from 'discord.js';
+import { Client, GatewayIntentBits, Events, AuditLogEvent } from 'discord.js';
 import { onInteraction } from './events/onInteraction';
 import { onReady } from './events/onReady';
 import { onMemberBan } from './events/onMemberBan';
+import { onMemberUnban } from './events/onMemberUnban';
 import { onMemberRemove } from './events/onMemberRemove';
 import { onMessageDelete } from './events/onMessageDelete';
 
 export const Bot = new Client({
-  intents: [Intents.FLAGS.GUILDS, Intents.FLAGS.GUILD_MESSAGES, Intents.FLAGS.GUILD_BANS, Intents.FLAGS.GUILD_MEMBERS],
+  intents: [
+    GatewayIntentBits.Guilds,
+    GatewayIntentBits.GuildMessages,
+    GatewayIntentBits.GuildModeration,
+    GatewayIntentBits.MessageContent,
+    GatewayIntentBits.GuildMembers,
+  ],
 });
 
 Bot.once('ready', async () => await onReady(Bot));
 
-Bot.on('guildMemberRemove', async (member) => await onMemberRemove(member));
-Bot.on('guildBanAdd', async (ban) => await onMemberBan(ban));
+Bot.on(Events.GuildAuditLogEntryCreate, async (auditLogEntry, guild) => {
+  if (auditLogEntry.action === AuditLogEvent.MemberBanAdd) {
+    await onMemberBan(auditLogEntry, guild);
+  } else if (auditLogEntry.action === AuditLogEvent.MemberBanRemove) {
+    await onMemberUnban(auditLogEntry, guild);
+  } else if (auditLogEntry.action === AuditLogEvent.MemberKick) {
+    await onMemberRemove(auditLogEntry, guild);
+  }
```

Comment:
switch would be a lot nicer here

Reply:
been doing too much python lately, will change to switch